controllers/datafoundation.go

```diff
@@ -195,3 +227,65 @@ func (r *dataFoundationReconciler) setDeviceSetCount(deviceSet *ocsv1.StorageDev
 		deviceSet.Count = currDeviceSetCount
 	}
 }
+
+func (r *dataFoundationReconciler) reconcileCSV() error {
+	r.Log.Info("Reconciling OCS Operator CSV")
+	csvList := opv1a1.ClusterServiceVersionList{}
+	if err := r.list(&csvList); err != nil {
+		return fmt.Errorf("unable to list csv resources: %v", err)
+	}
+	var csv *opv1a1.ClusterServiceVersion
+	for index := range csvList.Items {
+		obj := &csvList.Items[index]
+		if strings.HasPrefix(obj.Name, ocsOperatorName) {
+			csv = obj
+			break
+		}
+	}
+	if csv == nil {
+		r.Log.Info("unable to find OCS CSV")
+		return nil
+	}
+
+	isChanged := false
+	annotations := csv.GetAnnotations()
+	if value, found := annotations[managedFusionAnnotation]; !found ||
+		value != fmt.Sprintf("%s/%s", r.dfNamespace, r.dfName) {
+		r.addOwnerAnnotation(csv)
+		isChanged = true
+	}
+	deployments := csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs
+	for i := range deployments {
+		containers := deployments[i].Spec.Template.Spec.Containers
+		for j := range containers {
+			switch container := &containers[j]; container.Name {
+			case "ocs-operator":
+				resources := datafoundation.GetResourceRequirements("ocs-operator")
+				if !equality.Semantic.DeepEqual(container.Resources, resources) {
+					container.Resources = resources
+					isChanged = true
+				}
+			case "rook-ceph-operator":
+				resources := datafoundation.GetResourceRequirements("rook-ceph-operator")
+				if !equality.Semantic.DeepEqual(container.Resources, resources) {
+					container.Resources = resources
+					isChanged = true
+				}
+			case "ocs-metrics-exporter":
+				resources := datafoundation.GetResourceRequirements("ocs-metrics-exporter")
+				if !equality.Semantic.DeepEqual(container.Resources, resources) {
+					container.Resources = resources
+					isChanged = true
+				}
+			default:
+				r.Log.V(-1).Info("Could not find resource requirement", "Resource", container.Name)
+			}
```

Comment:
There is a repeated pattern here which obviously holds true for at least 3 cases, so I think we should collapse these 3 cases.
```suggestion
			switch container := &containers[j]; container.Name {
			case "ocs-operator",  "rook-ceph-operator",  "ocs-metrics-exporter":
				resources := datafoundation.GetResourceRequirements(container.Name)
				if !equality.Semantic.DeepEqual(container.Resources, resources) {
					container.Resources = resources
					isChanged = true
				}
			default:
				r.Log.V(-1).Info("Could not find resource requirement", "Resource", container.Name)
			}
```

And if the current assumption is that all containers will follow the same pattern then the preferred way is an `if` statement

```suggestion
			container := &containers[j]; container.Name 
			if container.Name == "ocs-operator" || container.Name == "rook-ceph-operator" || container.Name == "ocs-metrics-exporter" {
				resources := datafoundation.GetResourceRequirements(container.Name)
				if !equality.Semantic.DeepEqual(container.Resources, resources) {
					container.Resources = resources
					isChanged = true
				}
			else {
				r.Log.V(-1).Info("Could not find resource requirement", "Resource", container.Name)
			}
```

But it seems that the source of truth that derives this check is the CSV, not a static list which means a better solution might be
```suggestion
			container := &containers[j]; container.Name 
			resources := datafoundation.GetResourceRequirements(container.Name)
			if !equality.Semantic.DeepEqual(container.Resources, resources) {
				container.Resources = resources
				isChanged = true		
			}
```

As the assumption was always that we should panic and exist if we don't have a resource definition for a container and all containers should have a resource definition. That was done by design so we would capture these kinds of issues during development and not runtime!

Reply:
Okay, updated it